lib/line_collection.rb

```diff
@@ -228,8 +228,27 @@ def perform_word_wrap(text, width, first_line_number = 0, first_line_start = 0)
         else
           w, = $gtk.calcstringbox((line + word).rstrip, @size_enum, @font)
           if w > width
-            lines << Line.new(lines.length + first_line_number, first_line_start, line, true, @font, @size_enum)
-            first_line_start = lines.last.end
+            if line != ''
+              lines << Line.new(lines.length + first_line_number, first_line_start, line, true, @font, @size_enum)
+              first_line_start = lines.last.end
+            end
+
+            # break up long words
+            w, = $gtk.calcstringbox(word.rstrip, @size_enum, @font)
+            while w > width
+              n = word.length
+              while w > width
+                n -= 1
+                w, = $gtk.calcstringbox(word[0, n].rstrip, @size_enum, @font)
+                if w <= width
+                  lines << Line.new(lines.length + first_line_number, first_line_start, word[0, n], true, @font, @size_enum)
+                  first_line_start = lines.last.end
+                  word = word[n, word.length]
+                end
+              end
+
+              w, = $gtk.calcstringbox(word.rstrip, @size_enum, @font)
+            end
```

Comment:
This tanks perf with long words 😞

Reply:
Binary split might be worth trying... but of course that might be slower for words that are only slightly longer than the line..... 
You could try and distinguish between 
- line + word being very long (maybe `> 2*@w` ?) -> binary split
- less than that -> cut off char by char from behind (aka what you did)

Anyways let me also try a little refactor later tonight (want to try and extract the word wrap into a separate processing object - so it's easier to extract private methods and write more readable methods)

Feel free to revert it if you don't like it or find it too complex of course... but I want to at least try ;)

Now that we have tests it should be safe to do